packages/onchainkit/src/fund/hooks/useFundCardSetupOnrampEventListeners.ts

```diff
@@ -1,11 +1,18 @@
-import { useCallback, useEffect } from 'react';
+import { useCallback, useEffect, useRef } from 'react';
 import { useFundContext } from '../components/FundCardProvider';
 import { FUND_BUTTON_RESET_TIMEOUT } from '../constants';
 import type { EventMetadata, SuccessEventData } from '../types';
 import { setupOnrampEventListeners } from '../utils/setupOnrampEventListeners';
 
 export const useFundCardSetupOnrampEventListeners = () => {
   const { setSubmitButtonState, updateLifecycleStatus } = useFundContext();
+  const timeoutRef = useRef<NodeJS.Timeout | null>(null);
+
+  const scheduleFundButtonReset = useCallback(() => {
+    timeoutRef.current = setTimeout(() => {
+      setSubmitButtonState('default');
+    }, FUND_BUTTON_RESET_TIMEOUT);
+  }, [setSubmitButtonState]);
```

Comment:
I don't think this is likely to happen, but if `handleOnrampEvent` and `handleOnrampSuccess` are both called, and then the component unmounts, the cleanup function returned from `useEffect` will only clean up the latest timeout.

We could fix that by clearing the current timeout before setting a new one.